pkg/readiness/object_tracker.go

```diff
@@ -126,7 +135,30 @@ func (t *objectTracker) CancelExpect(o runtime.Object) {
 	delete(t.expect, k)
 	delete(t.seen, k)
 	delete(t.satisfied, k)
-	t.cancelled[k] = struct{}{}
+	t.cancelled[k] = mutatedObjData(t.mutators...)
+}
+
+func (t *objectTracker) TryCancelExpect(o runtime.Object) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	// Respect circuit-breaker.
+	if t.allSatisfied {
+		return
+	}
+
+	k, err := objKeyFromObject(o)
+	if err != nil {
+		log.Error(err, "skipping")
+		return
+	}
+
+	deleted := t.expect.decrementOrDelete(k)
+	if deleted {
+		delete(t.seen, k)
+		delete(t.satisfied, k)
+		t.cancelled[k] = mutatedObjData(t.mutators...)
```

Comment:
We probably want to leave this as an empty struct to avoid unnecessary memory usage.

Reply:
See my comment in other version of this comment.